### DISCUSSION TITLE
feature request: default nameplates

### DISCUSSION THREAD

**Discussion by Mrs-Feathers** [Zrips/CMI#1187] [Enhancement]
your plugin allows a command to set nameplates [names shown above a players head]
 however it doesn't allow you to se a default prefix. i would like to be able to set a default prefix to show ranks and stuff before a players name.. above their head. thank you!

**Comment by mercurialmusic** [comment]
I'd definitely vote for this one. We currently use NametagEdit, which adds the prefix well enough. But bonus points if you're able to do it with prefix and nickname (in color!). :)

**Reply by C4BR3R4** [reply]
+1

**Comment by Mrs-Feathers** [comment]
thats exactly what we use, and exactly the reason i'm asking for this as well...  it'd be awesome if it's all done with CMI... less plugins.. less lag.. everyone wins.

**Comment by Mrs-Feathers** [comment]
plus CMI already supports nameplates, why not make a "default" setting for it?

**Comment by kassadGLA** [comment]
[TAB REBORN](https://www.spigotmc.org/resources/tab-1-8-x-1-13-2-reborn.57806/)

**Comment by Mrs-Feathers** [comment]
> TAB REBORN

i dont think you read the issue. we already have a working plugin. we're asking for a CMI feature. to be able to have less plugins in our servers and since CMI already controls tablist too, then it should also control nameplates. which it does. but theres no "default option"... plus what you just linked me is for "tablists" only. i'm looking for nameplates.

**Comment by TomLewis** [comment]
+1
Rank based, For example I use ColouredTags at presnet and they have a groups.yml

```
deckhand: '&f[Deckhand] <name>'
default: '&f[Deckhand] <name>'
cadet: '&f[Cadet] <name> &3*'
sailor: '&7[Sailor]&f <name>'
```

But there is also a separate users one.

**Comment by hamcoremusic** [comment]
I *think* the reasoning behind not changing everyone's nameplates is if you even look at one of the most popular TAB plugins, it has to use armor stands to replace the name properly. This causes big client FPS hits for no real good reason.

It could be different for CMI, but that's my thinking.

**Comment by Mrs-Feathers** [comment]
most popular tab plugins just use ProtocolLib to edit username data being sent to the client connected to the server.

**Reply by hamcoremusic** [reply]
Ah okay. I'm curious if Zrips would want to make a ProtocolLib dependency though... As it is so off and on for updates and needing to explain to people they need a dev verisons all the time lol.

**Reply by TomLewis** [reply]
Most nearly every single decent server owner uses ProtocolLib for things like Anti Cheat, its not a big ask for a dependency @hamcoremusic

**Comment by Hazugg** [comment]
That would be great, I'm looking for the same, the command is good, but it would be great to be able to set it by default for groups as well as the tab list, but one of the nameplates would be great.
